langchain/vectorstores/pinecone.py

```diff
@@ -65,19 +66,27 @@ def add_texts(
             metadatas: Optional list of metadatas associated with the texts.
             ids: Optional list of ids to associate with the texts.
             namespace: Optional pinecone namespace to add the texts to.
+            batch_size: Optional batch size to use when embedding texts. Requires
+                embedding function to support batching.
 
         Returns:
             List of ids from adding the texts into the vectorstore.
 
         """
         # Embed and create the documents
-        docs = []
+        docs: list = []
         ids = ids or [str(uuid.uuid4()) for _ in texts]
+        # If a batch_size value is provided, we will batch the embeddings
+        if batch_size is not None:
+            embeddings = self._embedding_function(texts)
```

Comment:
do you have an example of how you would use this? most walkthroughs in the docs use Embedings.embed_query as the function, which takes in a single doc